datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -397,14 +454,111 @@ fn is_order_sensitive(aggr_expr: &Arc<dyn AggregateExpr>) -> bool {
         || aggr_expr.as_any().is::<ArrayAgg>()
 }
 
+/// Calculate the required input ordering for the [`AggregateExec`] by considering
```

Comment:
I know you didn't add it in this PR, but it seems to me like 

```rust
/// Checks whether the given aggregate expression is order-sensitive.
/// For instance, a `SUM` aggregation doesn't depend on the order of its inputs.
/// However, a `FirstAgg` depends on the input ordering (if the order changes,
/// the first value in the list would change).
fn is_order_sensitive(aggr_expr: &Arc<dyn AggregateExpr>) -> bool {
    aggr_expr.as_any().is::<FirstValue>()
        || aggr_expr.as_any().is::<LastValue>()
        || aggr_expr.as_any().is::<ArrayAgg>()
}
```

Might be better as a function of `AggregateExpr` (so that as people add new `AggregateExpr` they know there is some extra behavior used in the codebase) otherwise they will need to find this hard coded list

Reply:
I moved this utils function to [file](https://github.com/apache/arrow-datafusion/blob/main/datafusion/physical-expr/src/aggregate/mod.rs) where `AggregateExpr` trait is defined. Thanks for the suggestion